### PR TITLE
Codebraid: init at 0.5.0-unstable-2019-12-11

### PR DESCRIPTION
**Pull request by Synthetica9**
###### Motivation for this change

As to why I'm specifically using an unstable version: after 0.4.0 a few nice new features were introduced, like:

- [SageMath Support](https://github.com/gpoore/codebraid/commit/911e10325e36ca880af2bb3ae7d7227fba7ed55a)
- [Pandoc-Citeproc compatibility](https://github.com/gpoore/codebraid/commit/911e10325e36ca880af2bb3ae7d7227fba7ed55a)

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/python-modules/bespon/default.nix [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+  version = "0.3.0";
+  pname = "BespON";
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "0698vx1kh8c84f5qfhl4grdlyn1lljvdih8yczdz0pql8wkn8i7v";
+  };
+
+  propagatedBuildInputs = [ ];
+
```
Comment (jonringer) [inline pkgs/development/python-modules/bespon/default.nix]:
```suggestion
  # upstream doesn't contain tests
  doCheck = false;
  
  pythonImportsCheck = [ "bespon" ];
```

File: pkgs/tools/misc/codebraid/default.nix [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+  src = fetchFromGitHub {
+    owner = "gpoore";
+    repo = pname;
+    rev = "fac1b29";
+    sha256 = "0ldfrkkip7i1fdyz1iydyik3mhm0xv0jvxnl37r7g707dl38vf3h";
+  };
+
+  propagatedBuildInputs = with python3Packages; [ bespon ];
+
```
Comment (jonringer) [inline pkgs/tools/misc/codebraid/default.nix]:
```suggestion
  # unfortunately upstream doesn't contain tests
  checkPhase = ''
    $out/bin/codebraid --help > /dev/null
  '';
```

**Review by jonringer — COMMENTED** [review]
I would prefer unittests, but I'll compromise for the suggestions.

File: pkgs/tools/misc/codebraid/default.nix (hunk 2 above)
Comment (Synthetica9) [reply pkgs/tools/misc/codebraid/default.nix]: codebraid: add trivial check in lieu of unit tests

**Comment by Synthetica9** [comment]
@jonringer Request granted :)

**Comment by jonringer** [comment]
please squash the fix up commits into the related commits, `git rebase -i` makes this easy

**Comment by Synthetica9** [comment]
@jonringer Done

**Review by jonringer — APPROVED** [review]
LGTM, would still like more tests

```
[7 built, 0.0 MiB DL]
https://github.com/NixOS/nixpkgs/pull/80536
4 package built:
codebraid python27Packages.bespon python37Packages.bespon python38Packages.bespon
```

**Comment by jonringer** [comment]
@GrahamcOfBorg build codebraid python27Packages.bespon python37Packages.bespon python38Packages.bespon

**Pull request merged**